main.cpp

```diff
@@ -4592,7 +4602,9 @@ bool load_guts(picoboot::connection con, iostream_memory_access &file_access) {
         // new scope for progress bar
         {
             progress_bar bar("Loading into " + memory_names[type] + ": ");
-            uint32_t batch_size = FLASH_SECTOR_ERASE_SIZE;
+            // Use batches of size/100 rounded up to FLASH_SECTOR_ERASE_SIZE
+            uint32_t batch_size = std::max(FLASH_SECTOR_ERASE_SIZE,
```

Comment:
arguably the std::max is unnecessary, since everything but 0 would be rounded up anyway